### PR TITLE
Add contributing template

### PR DESCRIPTION
**Pull request by dingo-d**
Added the contributing template.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+## Public properties
+
+When writing sniffs, always remember that any `public` sniff property can be overruled via a custom ruleset by the end-user.
+Only make a property `public` if that is the intended behaviour.
+
+When you introduce new `public` sniff properties, or your sniff extends a class from which you inherit a `public` property, please don't forget to update the [public properties wiki page](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties) with the relevant details once your PR has been merged into the `develop` branch.
+
+# Unit Testing
+
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: "_All PRs which affect sniffs, whether bug fixes to existing sniffs, or the addition of a new sniff, should be accompanied by unit tests._"

~~Maybe we should even have a little section about sniff names, categories, unit test case files and unit test files in general.~~ Found it! (see below)

Aside from a section on how to categorize a new sniff in the first place, seeing how difficult that turned out to be in the original categorization.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+composer install
+```
+
+will install PHP_CodeSniffer, WordPress Coding Standards, PHPUnit, PHPCompatibility and security advisories which ensures that your application doesn't have installed dependencies with known security vulnerabilities.
+
+### Other setups
+
+If you have PHP_CodeSniffer (PHPCS) and/or WordPress Coding Standards (WordPressCS) already installed, for instance as git clones, because you either contribute to these libraries as well or you want to develop for or test WPThemeReview with bleeding edge versions of either PHP_CodeSniffer or WordPress Coding Standards, you need to take some additional steps to make it all work.
+
+First, make sure you also have PHPCompatibility and make sure the `installed_paths` for PHP_CodeSniffer is setup correctly.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: Maybe add a link to the PHPCS wiki page with the info on setting `installed_paths` ?

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+#### Method 1
+
+Set up environment variables to point to PHP_CodeSniffer and WordPress Coding Standards:
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: Do we need to explain these would be OS/system-wide environment variables ?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+```sh
+PHPCS_DIR: I:/000_GitHub/PHPCS/PHP_CodeSniffer/
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: Please anonymize these paths, like `/path/to/PHP_CodeSniffer/`

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+WPCS_DIR: I:/000_GitHub/PHPCS/WordPressCS/
+```
+
+If you do that, everything should work as expected.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: `everything` => `the unit tests should be able to work correctly`

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    <env name="PHPCS_DIR" value="path/to/PHP_CodeSniffer/"/>
+    <env name="WPCS_DIR" value="path/to/WordPressCS/"/>
+  </php>
+```
+
+(and don't remove the existing line within the `<php>` block.)
+
+## Writing and running unit tests
+
+The most iportant thing when writing sniffs intended for the theme review, is to have an ample of examples. This makes writing sniffs a lot easier, because you can test agains the give nexamples.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: `most iportant thing` => `most important thing`

`an ample of examples` => `ample example code`

`give nexamples` => `given examples`

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+............                                                      12 / 12 (100%)
+
+Tests generated 57 unique error codes; 0 were fixable (0%)
+
+Time: 13.55 seconds, Memory: 64.00MB
+
+OK (12 tests, 0 assertions)
+```
+
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: "_If you didn't install PHPCS/WPCS/PHPUnit using Composer, you will need to type the above command in to run the unit tests. Make sure you replace the path to PHPUnit and the path to PHPCS when you do._"

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+## Unit Testing conventions
+
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: *grin*, ok, so apparently I expected this bit higher up in the file (see my earlier remark). We may need to have a think about the order of the various sections.

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+You can also run the same command using composer scripts
+
+```bash
+composer check-cs -- --standard=WPThemeReview -s WPThemeReview/Tests/CoreFunctionality/FileIncludeUnitTest.inc --sniffs=WPThemeReview.CoreFunctionality.FileInclude
+```
+
+## Code Standards for this project
+
+The sniffs and test files - not test _case_ files! - for WPCS should be written such that they pass the `WordPress-Extra` and the `WordPress-Docs` code standards using the custom ruleset as found in `/.phpcs.xml.dist`.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: This needs to be adjusted to a more WPTRCS specific text + could mention that to test this you can run the `composer check-cs` command from the project root.

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+Once a commit is made to `develop`, a PR should be opened from `develop` into `master` and named "Next release". This PR will provide collaborators with a forum to discuss the upcoming stable release.
+
+# Considerations when writing sniffs
+
+## Public properties
+
+When writing sniffs, always remember that any `public` sniff property can be overruled via a custom ruleset by the end-user.
+Only make a property `public` if that is the intended behaviour.
+
+When you introduce new `public` sniff properties, or your sniff extends a class from which you inherit a `public` property, please don't forget to update the [public properties wiki page](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties) with the relevant details once your PR has been merged into the `develop` branch.
```
[the remark this answers is not in the document]
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: If this is (or will be) written in the WPCS repo, maybe it wouldn't be a bad idea to just link to the part in the WPCS repos contributing template? Something like:

```
If you want to know a bit more about writing sniffs, check out the link in the wpcs repository with the detailed instructions.

[Link to instructions](link)
```

File: .github/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+
+Bug reports containing a minimal code sample which can be used to reproduce the issue are highly appreciated as those are most easily actionable.
+
+# Contributing patches and new features
+
+## Branches
+
+Ongoing development will be done in the `develop` branch with merges done into `master` once considered stable.
+
+To contribute an improvement to this project, fork the repo and open a pull request to the `develop` branch. Alternatively, if you have push access to this repo, create a feature branch prefixed by `feature/` and then open an intra-repo PR from that branch to `develop`.
```
[the remark this answers is not in the document]
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: I'm fine with `feature/` branch convention. I'm using it in other projects. But if it's easier for you, we can agree on some other convention. Like `new-sniff/sniff-name` or `update-sniff/sniff-name`.

File: .github/CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+* PHP 5.4 or higher
+* WordPress-Coding-Standards 1.0.0 or higher
+* PHP_CodeSniffer 3.3.0 or higher
+* PHPUnit 4.x, 5.x, 6.x or 7.x
+* PHPCompatibility 8.0 or higher
+
+### Composer setup
+
+Use Composer to install all the necessary dependencies to write and run unit tests. Running
+
```
[the remark this answers is not in the document]
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: To me, it's self-explanatory since the `composer.json` is in the project root, but I can add it for someone who isn't acquainted with Composer :+1:

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: I can add `global` in front of `environment variables` to point that out

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: Is this ok?

```bash
PHPCS_DIR: I:/path/to/PHP_CodeSniffer/
WPCS_DIR: I:/path/to/WordPressCS/
```

File: .github/CONTRIBUTING.md (hunk 8 above)
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: I can use 

```md
<details>
<summary>Click to toggle contents of `code`</summary>
    ```
    CODE!
    ```
</details>
```

Which should toggle the part in the `<details>` tag. Or we can agree first on a general titles and subtitles organization and then use it.

File: .github/CONTRIBUTING.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: Oh I get it. This should be moved down where the writing the sniff is, and we could put this in the `<details>` tags so that it can be expanded.

We can also then use this part in the readme (collapsed)

**Comment by dingo-d** [comment]
I've updated the contributing template with the recommended changes and I've also folded the code of the sniff file so that the entire file isn't so long. 

I'm waiting on the second round of recommendations :D

File: .github/CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+You can see how this can be done by reading the official PHPCS [documentation](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#setting-the-installed-standard-paths).
+
+#### Method 1
+
+Set up global environment variables to point to PHP_CodeSniffer and WordPress Coding Standards:
+
+```bash
+PHPCS_DIR: I:/path/to/PHP_CodeSniffer/
+WPCS_DIR: I:/path/to/WordPressCS/
```
Comment (ntwb) [inline .github/CONTRIBUTING.md]: Why is this required? Shouldn't the _dealerdirect/phpcodesniffer-composer-installer_ package take care of this path?

Comment (jrfnl) [reply .github/CONTRIBUTING.md]: Please read the whole document. A composer-based setup is the default, however, for devs, other setups are also possible which allow for testing with, for instance, dev branches of WPCS or PHPCS.
This section explains how to get the unit test working in that case.

**Comment by dingo-d** [comment]
I've posted on twitter, and on AWP group on facebook, and several times on the themereview channel in slack, and didn't get much feedback. Is this good to go?

**Comment by dingo-d** [comment]
I've posted on twitter, and on AWP group on facebook, and several times on the themereview channel in slack, and didn't get much feedback.

A colleague of mine suggested that we add a TL;DR version, but I'm not sure how to sumarize this into a short format to be honest.

Can we merge this, or do we want to wait? It would be good to do this as soon as possible because of Twentynineteen theme 🙂

**Comment by joyously** [comment]
@dingo-d Those of us that see your call for feedback don't know enough to give any. We need the contributing document, so can't contribute to it...

**Comment by dingo-d** [comment]
@joyously The point is to go through the template and see if you can understand how to contribute to the project. Kinda like the first step to contributing 😉 

@jrfnl I'll push the TOC and see if it works, and then do the rebasing, but I'll either have to do it late today, or tomorrow morning.

I'm also fine with merging it after this, and then making the changes after that 🙂

**Comment by dingo-d** [comment]
@jrfnl I think I messed something up when I attempted the rebasing :S Should I try to revert the changes? I really suck at rebasing :/

File: .github/CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+    + [File organization and naming](#file-organization-and-naming)
+  * [Code Standards for this project](#code-standards-for-this-project)
+
+# Reporting Bugs
+
+When reporting a bug in an existing sniff it's good to differentiate on whether the sniff is reporting something which the sniff shouldn't report on (a _false positive_), or whether the sniff isn't reporting on a certain thing that it should report on (a _false negative_).
+
+In the case you found something that a sniff shouldn't report on, you can check where it's coming from using the `-s` flag with the `phpcs` command.
+
+In case the sniff is not reporting on a certain thing it should, you won't be able to check it using the `-s` flag, but **in both** cases it's **mandatory** to provide code samples so that the issue could be easily remedied.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
In case of a false negative, you won't be able to check for a sniff code using the `-s` flag, but **in both** cases it's **mandatory** to provide code samples so that the issue can be easily reproduced and remedied.
```

File: .github/CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+
+## Upstream bugs
+
+In the case the sniff error code doesn't starts with `WPThemeReview`, and instead it starts with `WordPress`, `PHPCompatibility`, or something else, that means that it is and 'upstream' bug coming from either `WPCS`, `PHPCompatibility` or `PHPCS`. You can report the bug here, but the chances are high that you'll be asked to report it in the correct repository instead.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
If the sniff error code doesn't starts with `WPThemeReview`, but instead it starts with `WordPress`, `PHPCompatibility`, or something else, that means that it is an 'upstream' bug coming from either [`WordPressCS`](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards), [`PHPCompatibility`](https://github.com/PHPCompatibility/PHPCompatibility) or [`PHP_CodeSniffer`](https://github.com/squizlabs/PHP_CodeSniffer).

You can report the bug here, but the chances are high that you'll be asked to report it in the correct repository instead.
```

File: .github/CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,259 @@
...
+
+# Contributing patches and new features
+
+## Branches
+
+Ongoing development will be done in the `develop` branch with merges done into `master` once considered stable.
+
+## Picking an open issue
+
+If you want to pick an open issue, please mention that in the issue, and it will be assigned to you. This will prevent the possible double work by different people on the same issue.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
If you start work on an open issue, please mention that in the issue, and it will be assigned to you. This will prevent possible double work by different people on the same issue.
```

File: .github/CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,259 @@
...
+
+## Contributing to the project
+
+If you want to contribute to this project, fork the repo and open a pull request to the `develop` branch. If you have push access to this repo, you can create a feature branch and then open an intra-repo PR from that branch to the `develop`.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
If you want to contribute to this project, fork the repo, create a new branch for your work and, once finished, open a pull request to the `develop` branch. If you have push access to this repo, you don't need to fork the repo, but can push your feature branch directly to this repo and you can then open an intra-repo PR from that branch to the `develop` branch.
```

File: .github/CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,259 @@
...
+
+In the case the sniff error code doesn't starts with `WPThemeReview`, and instead it starts with `WordPress`, `PHPCompatibility`, or something else, that means that it is and 'upstream' bug coming from either `WPCS`, `PHPCompatibility` or `PHPCS`. You can report the bug here, but the chances are high that you'll be asked to report it in the correct repository instead.
+
+# Contributing patches and new features
+
+## Branches
+
+Ongoing development will be done in the `develop` branch with merges done into `master` once considered stable.
+
+## Picking an open issue
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: Suggestion: move this subsection to below the `Contributing with code` subsection.

File: .github/CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,259 @@
...
+
+If you want to pick an open issue, please mention that in the issue, and it will be assigned to you. This will prevent the possible double work by different people on the same issue.
+
+## Contributing to the project
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
## Contributing with code
```

File: .github/CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,259 @@
...
+
+All PRs which affect sniffs, whether bug fixes to existing sniffs, or the addition of a new sniff, should be accompanied by unit tests.
+
+### Pre-requisites
+
+* PHP 5.4 or higher
+* WordPress-Coding-Standards 1.0.0 or higher
+* PHP_CodeSniffer 3.3.0 or higher
+* PHPUnit 4.x, 5.x, 6.x or 7.x
+* PHPCompatibility 8.0 or higher
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
* PHPCompatibility 9.0 or higher
```

File: .github/CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,259 @@
...
+
+### Composer setup
+
+Use Composer to install all the necessary dependencies to write and run unit tests. Running
+
+```bash
+composer install
+```
+
+from the root of the cloned repository, will install `PHP_CodeSniffer`, `WordPress Coding Standards`, `PHPUnit`, `PHPCompatibility` and `security advisories` which ensures that your application doesn't have installed dependencies with known security vulnerabilities.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
from the root of the cloned repository, will install `PHP_CodeSniffer`, `WordPress Coding Standards`, `PHPUnit`, `PHPCompatibility` and `security advisories`, which ensures that you won't install dependencies with known security vulnerabilities.
```

File: .github/CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,259 @@
...
+
+# Considerations when writing sniffs
+
+## Unit Testing
+
+All PRs which affect sniffs, whether bug fixes to existing sniffs, or the addition of a new sniff, should be accompanied by unit tests.
+
+### Pre-requisites
+
+* PHP 5.4 or higher
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: I think we can remove the pre-requisites, except for the PHP requirement.

Everything else will normally be installed via Composer, so is not a prerequisite.

File: .github/CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,259 @@
...
+composer install
+```
+
+from the root of the cloned repository, will install `PHP_CodeSniffer`, `WordPress Coding Standards`, `PHPUnit`, `PHPCompatibility` and `security advisories` which ensures that your application doesn't have installed dependencies with known security vulnerabilities.
+
+### Other setups
+
+If you have PHP_CodeSniffer (PHPCS) and/or WordPress Coding Standards (WordPressCS) already installed, for instance as git clones, because you either contribute to these libraries as well or you want to develop for or test WPThemeReview with bleeding edge versions of either PHP_CodeSniffer or WordPress Coding Standards, you need to take some additional steps to make it all work.
+
+First, make sure you also have PHPCompatibility and make sure the `installed_paths` for PHP_CodeSniffer is setup correctly.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
First, make sure you also have PHPCompatibility installed and make sure the `installed_paths` for PHP_CodeSniffer is setup correctly.
```

File: .github/CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,259 @@
...
+
+```xml
+  <php>
+    <env name="PHPCS_DIR" value="path/to/PHP_CodeSniffer/"/>
+    <env name="WPCS_DIR" value="path/to/WordPressCS/"/>
+  </php>
+```
+
+(and don't remove the existing line within the `<php>` block.)
+
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: Add: `Once you've done that, both running the sniffs as well as the unit tests should work correctly.`

File: .github/CONTRIBUTING.md [hunk 25]
```diff
@@ -0,0 +1,259 @@
...
+#### Method 1
+
+Set up global environment variables to point to PHP_CodeSniffer and WordPress Coding Standards:
+
+```bash
+PHPCS_DIR: I:/path/to/PHP_CodeSniffer/
+WPCS_DIR: I:/path/to/WordPressCS/
+```
+
+If you do that, the unit tests should be able to work correctly.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
If you do that, both the sniffs as well as the unit tests should be able to work correctly.
```

File: .github/CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,259 @@
...
+  </php>
+```
+
+(and don't remove the existing line within the `<php>` block.)
+
+## Writing and running unit tests
+
+The most important thing when writing sniffs intended for the theme review, is to have ample example code. This makes writing sniffs a lot easier, because you can test against the given examples.
+
+If you want to run unit tests, and if you ran `composer install`, you can now run `composer run-tests` (or `composer run-script run-tests`), which will run the test suite.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
If you want to run the unit tests, and you installed the dependencies using `composer`, you can now run `composer run-tests` (or `composer run-script run-tests`), which will run the test suite.
```

File: .github/CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,259 @@
...
+
+The WordPress Theme Review Coding Standards use the PHP_CodeSniffer native unit test suite for unit testing the sniffs.
+
+Once you've started the tests you will see a similar output to this:
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
Once you've started the tests you will see output similar to this:
```

File: .github/CONTRIBUTING.md [hunk 28]
```diff
@@ -0,0 +1,259 @@
...
+Time: 13.55 seconds, Memory: 64.00MB
+
+OK (12 tests, 0 assertions)
+```
+
+If you didn't install PHPCS/WPCS/PHPUnit using Composer, you will need to type the above command in to run the unit tests. Make sure you replace the path to PHPUnit and the path to PHPCS when you do.
+
+## Unit Testing conventions
+
+### Public properties
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: This section should be moved up to the `Contributing patches and new features` section as it is about writing sniffs, not about unit testing.

File: .github/CONTRIBUTING.md [hunk 29]
```diff
@@ -0,0 +1,259 @@
...
+        8 => 1,
+      );
+    }
+
+  }
+  ```
+  
+</details><br>  
+
+Also note the class name convention. The method `getWarningList()` MUST return an array of line numbers indicating warnings (when running `phpcs`) found in `WPThemeReview/Tests/CoreFunctionality/FileIncludeSniff.inc`.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
Also note the class and function name conventions. The methods `getErrorList()` and `getWarningList()` MUST return either an empty array (no errors/warnings) or an array of line numbers indicating errors/warnings (when running `phpcs`) found in the test case file for the sniff.
```

File: .github/CONTRIBUTING.md [hunk 30]
```diff
@@ -0,0 +1,259 @@
...
+If you run
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
If you run:
```

File: .github/CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,259 @@
...
+
+```bash
+$ cd /path-to-cloned/WPThemeReview
+$ vendor/bin/phpcs --standard=WPThemeReview -s WPThemeReview/Tests/CoreFunctionality/FileIncludeUnitTest.inc --sniffs=WPThemeReview.CoreFunctionality.FileInclude
+```
+
+The results will be
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
The results should be:
```

File: .github/CONTRIBUTING.md [hunk 32]
```diff
@@ -0,0 +1,259 @@
...
+
+```bash
+...
+FILE: /Users/infinum-denis/Projects/Personal/WPThemeReview/WPThemeReview/Tests/CoreFunctionality/FileIncludeUnitTest.inc
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
FILE: /path-to-cloned/WPThemeReview/WPThemeReview/Tests/CoreFunctionality/FileIncludeUnitTest.inc
```

File: .github/CONTRIBUTING.md [hunk 33]
```diff
@@ -0,0 +1,259 @@
...
+
+You can also run the same command using composer scripts
+
+```bash
+composer check-cs -- --standard=WPThemeReview -s WPThemeReview/Tests/CoreFunctionality/FileIncludeUnitTest.inc --sniffs=WPThemeReview.CoreFunctionality.FileInclude
+```
+
+## Code Standards for this project
+
+The WPTRTCS sniffs and test files (excluding test _case_ files) are written in a way that they pass the rules set by the custom ruleset found in `/.phpcs.xml.dist`. They should pass some of the `WordPress-Extra` standards and the `WordPress-Docs` code standards.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
The WPTRTCS sniffs and test files (excluding test _case_ files) are written in a way that they pass the rules set by the custom ruleset found in [`/.phpcs.xml.dist`](https://github.com/WPTRT/WPThemeReview/blob/develop/.phpcs.xml.dist). They should pass some of the `WordPress-Extra` standards and the `WordPress-Docs` code standards.
```

File: .github/CONTRIBUTING.md [hunk 34]
```diff
@@ -0,0 +1,259 @@
...
+
+You can check the custom written sniff using the `composer check-cs` command from the project root.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
You can check whether your code complies with the coding standard using the `composer check-cs` command from the project root.
```

File: .github/CONTRIBUTING.md [hunk 35]
```diff
@@ -0,0 +1,259 @@
...
+
+The `--sniffs=...` directive limits the output to the sniff you are testing.
+
+You can also run the same command using composer scripts
+
+```bash
+composer check-cs -- --standard=WPThemeReview -s WPThemeReview/Tests/CoreFunctionality/FileIncludeUnitTest.inc --sniffs=WPThemeReview.CoreFunctionality.FileInclude
+```
+
+## Code Standards for this project
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: It may not be a bad idea to move this up and make this a subsection of `Contributing patches and new features`.

**Comment by dingo-d** [comment]
@jrfnl I've made the changes you suggested. Thanks again for fixing all the issues :)

**Comment by jrfnl** [comment]
@dingo-d Not sure what happened here, but it looks like the rebase we did earlier this week has been undone now.

Want me to re-do it ? (and squash those last commits)

**Comment by dingo-d** [comment]
@jrfnl If you'll have the time, can you rebase and merge if everything is ok? Thanks 🙂

**Comment by jrfnl** [comment]
@dingo-d Looks like it's that last "Reorder" commit which merges the rebased branch back to the old branch....

File: .github/CONTRIBUTING.md [hunk 36]
```diff
@@ -0,0 +1,257 @@
...
+composer install
+```
+
+from the root of the cloned repository, will install `PHP_CodeSniffer`, `WordPress Coding Standards`, `PHPUnit`, `PHPCompatibility` and `security advisories`, which ensures that you won't install dependencies with known security vulnerabilities.
+
+### Other setups
+
+If you have PHP_CodeSniffer (PHPCS) and/or WordPress Coding Standards (WordPressCS) already installed, for instance as git clones, because you either contribute to these libraries as well or you want to develop for or test WPThemeReview with bleeding edge versions of either PHP_CodeSniffer or WordPress Coding Standards, you need to take some additional steps to make it all work.
+
+First, make sure you also have PHPCompatibility installed and make sure the `installed_paths` for PHP_CodeSniffer is setup correctly.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]: Darn... missed this one last time round.
```suggestion
First, make sure you also have PHPCompatibility installed and make sure the `installed_paths` for PHP_CodeSniffer is set up correctly.
```

File: .github/CONTRIBUTING.md [hunk 37]
```diff
@@ -0,0 +1,257 @@
...
+  </php>
+```
+
+(and don't remove the existing line within the `<php>` block.)
+
+Once you've done that, both running the sniffs as well as the unit tests should work correctly.
+
+## Writing and running unit tests
+
+The most important thing when writing sniffs intended for the theme review, is to have ample example code. This makes writing sniffs a lot easier, because you can test against the given examples.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md]:
```suggestion
The most important thing when writing sniffs intended for the theme review standard, is to have ample example code. This makes writing sniffs a lot easier, because you can test against the given examples.
```

File: .github/CONTRIBUTING.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (dingo-d) [reply .github/CONTRIBUTING.md]: Yeah, looking at it this paragraph makes no sense from the WPThemeReview point of view. We don't have a wiki, and we can document any changes when adding new sniff in the changelog. And maybe add a wiki down the line.

**Comment by jrfnl** [comment]
Great job @dingo-d !
